src/Training.AirBnb.Clone.Backend/Backend Project/Domain/Services/AmenityCategoryService.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Backend_Project.Domain.Entities;
+using Backend_Project.Domain.Exceptions.AmentyCategoryException;
+using Backend_Project.Domain.Interfaces;
+using Backend_Project.Persistance.DataContexts;
+using System.Linq.Expressions;
+
+namespace Backend_Project.Domain.Services
+{
+    public class AmenityCategoryService : IEntityBaseService<AmenityCategory>
+    {
+        private readonly IDataContext _appDataContext;
+
+        public AmenityCategoryService(IDataContext appDataContext)
+        {
+            _appDataContext = appDataContext;
+        }
+
+        public async ValueTask<AmenityCategory> CreateAsync(AmenityCategory amenityCategory, bool saveChanges = true, CancellationToken cancellationToken = default)
+        {
+            if (!IsValidCategoryName(amenityCategory.CategoryName))
+                throw new AmenityCategoryFormatException("Invalid categoryName!");
+
+            await _appDataContext.AmenityCategories.AddAsync(amenityCategory, cancellationToken);
+
+            if (saveChanges)
+                await _appDataContext.AmenityCategories.SaveChangesAsync();
+
+            return amenityCategory;
+        }
+
+        public async ValueTask<AmenityCategory> DeleteAsync(Guid id, bool saveChanges = true, CancellationToken cancellationToken = default)
+        {
+            var deletedAmenityCategory = await GetByIdAsync(id);
+
+            if (deletedAmenityCategory is null)
+                throw new AmenityCategoryNotFoundException("AmentyCategory not found!");
+
```

Comment:
GetByIdAsync qilib object olgandan keyin uni null ga tekshirish shart emas. Chunki GetByIdAsync ichida topilmasa exception throw qiladi va bu methodda yana tekshirib o'tirish shartmas. Qolgan methodlarda ham to'g'irlab chiqing.